torch/distributed/_tensor/api.py

```diff
@@ -57,6 +57,9 @@ class _ToTorchTensor(torch.autograd.Function):
     @staticmethod
     def forward(ctx, input: "DTensor"):  # type: ignore[override]
         ctx.dtensor_spec = input._spec
+        # detach local tensor from its autograd graph as autograd works
```

Comment:
from our offline discussion, we might want to change this to a view_as() to avoid any confusion on the autograd implication of the detach here (which is none!).